### PR TITLE
cppdap: add recipe

### PR DESCRIPTION
**Pull request by gjasny**
Specify library name and version:  **cppdap/cci.20230526**

CMake 3.27 will have `cppdap` as a dependency (see [merge request](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8338))

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by memsharded — APPROVED** [review]

File: recipes/cppdap/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["cppdap"]
+
+        self.cpp_info.set_property("cmake_file_name", "cppdap")
+        self.cpp_info.set_property("cmake_target_name", "cppdap::cppdap")
```
Comment (memsharded) [inline recipes/cppdap/all/conanfile.py:94]: When this matches the package name, neither of these to properties are necessary, they are already the default.
No need to change it, but just you know it, for simpler recipes.

Comment (gjasny) [reply recipes/cppdap/all/conanfile.py:94]: Thanks for sharing!

File: recipes/cppdap/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        elif self.options.with_json == "rapidjson":
+            self.requires("rapidjson/cci.20220822")        
+        elif self.options.with_json == "nlohmann_json":
+            self.requires("nlohmann_json/3.11.2")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        if self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
```
Comment (SpaceIm) [inline recipes/cppdap/all/conanfile.py:62]: if it raises for shared, why providing shared option? And why message refers to Visual Studio and msvc specifically while it raises regardless of compiler?

Comment (SpaceIm) [reply recipes/cppdap/all/conanfile.py:62]: FYI, shared build is allowed since https://github.com/google/cppdap/pull/116

**Pull request merged**